### ISSUE TITLE
bosh-init deploy vCloud error

### ISSUE THREAD

**Issue by Prospecta** [cloudfoundry/bosh#1436]
Hi,

I am encountering an error when attempting to run the bosh-init deploy script on vCloud (not vCloud Air).  

The job fails when attempting to create the vm:
---

Command 'deploy' failed:
  Deploying:
    Creating instance 'bosh/0':
      Creating VM:
        Creating vm with stemcell cid 'urn:vcloud:catalogitem:decd3c65-5dde-47ae-836e-100199cdae0d':
          CPI 'create_vm' method responded with error: CmdError{"type":"Unknown","message":"Task urn:vcloud:task:14f18678-7d9a-4046-a1aa-d708c8058be0 Updated Virtual Machine 3b6b2f67-572c-44c4-74d4-a80b28887836(be1c153c-6e01-46b3-81c5-018ec2c26c10) completed unsuccessfully, Details:  [ 01b88b27-f995-41dd-b39e-c2d498b808fb-3776 ] Unable to perform this action. Contact your cloud admi...","ok_to_retry":false}
---

and vCloud returns the error:
---

com.vmware.ssdc.library.exceptions.DatastoreNotAvailableException: null
    at com.vmware.vcloud.fabric.storage.placement.sdrs.impl.SdrsPlacementManagerImpl.processSdrsResults(SdrsPlacementManagerImpl.java:1040)
    at com.vmware.vcloud.fabric.storage.placement.sdrs.impl.SdrsPlacementManagerImpl.selectDatastoreInStoragePod(SdrsPlacementManagerImpl.java:213)
    at com.vmware.vcloud.fabric.storage.placement.impl.VirtualMachineDiskLevelStorageSelectorImpl.selectDatastore(VirtualMachineDiskLevelStorageSelectorImpl.java:369)
    at com.vmware.vcloud.fabric.storage.storedVm.impl.RelocateStoredVmByStorageClassActivity$CalculateTargetDatastorePhase.invoke(RelocateStoredVmByStorageClassActivity.java:138)
    at com.vmware.vcloud.activity.executors.ActivityRunner.runPhase(ActivityRunner.java:156)
    at com.vmware.vcloud.activity.executors.ActivityRunner.run(ActivityRunner.java:118)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
---

The vClouderror reports that the datastore is not available when in fact there is a valid datastore available for the deployment which has no issues when creating a vm manually via vCloud Director.  The stemcell is also uploaded to the datastore with no issues.

We had previously deployed Cloud Foundry v182 including Bosh with no trouble on vCloud but now struggle with the latest releases (bosh v257.9 & vCloud CPI v24).  We have attempted to use both the latest vCloud and vSphere stemcells (v3262.12) with no luck.

For reference, below is the vcd section of the bosh.yml:
---

```
vcd: &vcd # <--- Replace values below
  url: https://*******************
  user: **********
  password: ************
  entities:
    organization: Cloud-Foundry
    virtual_datacenter: Cloud-Foundry-VDC
    vapp_catalog: cf-catalog-bosh
    media_catalog: cf-catalog-bosh
    media_storage_profile: '*'
    vm_metadata_key: bosh-meta
  control: {wait_max: 900}
```
---

Any ideas what could be causing the issue?

Thanks

**Comment by poblin-orange** [comment]
Hello,
also using vcloud with bosh (oldest version though)
Did u try specifying the exact vapp_storage_profile, as found in vcloud director portal ?
Here is our config :
```
    entities:
      organization: Cube4PaasProd
      virtual_datacenter: Apps-Dev
      vapp_catalog: bosh
      media_catalog: bosh
      media_storage_profile: "Standard Performances"
      vapp_storage_profile: "Standard Performances"
      vm_metadata_key: cf-agent-env
      description: vcd-cf
```

**Comment by Prospecta** [comment]
Hi,

Attempted it again with an exact vapp_storage_profile but still receiving the same error.  I have attached the debug log in case there's anything obvious in there.

Thanks
[bosh.log.zip](https://github.com/cloudfoundry/bosh/files/473350/bosh.log.zip)

**Comment by poblin-orange** [comment]
Hello,
nothing obvious in your logs. I would guess theres is something wrong with the vcloud / vsphere configuration for independant disk. The bosh persistent disk is created with a vcloud independent disk, which you cant create from the vcloud portal to troubleshoot.

**Comment by Prospecta** [comment]
Hi,

Any idea how I can determine that the problem is related to independent disks if I can't create them from the vCloud portal?

Thanks

**Comment by poblin-orange** [comment]
Unfortunatly, no easy way.
The vcloud API is restfull, but very complex to use (SOAP payload).
I guess you cloud try creating a simple ruby / fog script (ie: create independent disk, attach to an exisiting vm)...

**Comment by jleavers** [comment]
Perhaps these GDS utilities would help here -
http://gds-operations.github.io/vcloud-tools/



On 15 September 2016 at 11:56:13, Pierre Oblin (notifications@github.com)
wrote:

Unfortunatly, no easy way.
The vcloud API is restfull, but very complex to use (SOAP payload).
I guess you cloud try creating a simple ruby / fog script (ie: create
independent disk, attach to an exisiting vm)...

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/cloudfoundry/bosh/issues/1436#issuecomment-247297415>,
or mute the thread

**Comment by poblin-orange** [comment]
@jleavers thks, didnt know about these tools .

**Comment by Prospecta** [comment]
Hi,

So I've been engaging with VMware support this week and they suggested we switch off SDRS for the datastore cluster that Bosh is being deployed to.  Apparently there is an SDRS placement issue which affects version 8.0.1 of vCloud.

I have attempted the deployment of Bosh again (with SDRS turned off) and it has run successfully.

I have just posed the question to VMware to understand the impact of turning off this feature when deploying additional VMs to the datastore cluster.  I'll let you know what they come back with.

Cheers
